### PR TITLE
Add allow_overlap and allow empty spheres in sphere aggregation

### PR DESCRIPTION
**Pull request by fraimondo**
Features:
* Allow for empty spheres (this will result in arrays with the second dimention of size 0). Applying any aggregation function should yield a NaN.
* Expose `allow_overlap` parameter to every sphere-based marker.

* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

### CODE REVIEW

File: docs/changes/latest.inc [hunk 1]
```diff
@@ -51,6 +51,12 @@ Enhancements
 
 - Add ``pre_run`` parameter to ``_queue_condor`` (:gh:`188` by `Fede Raimondo`_).
 
+- Allow for empty spheres in :class:`junifer.external.nilearn.JuniferNiftiSpheresMasker`, that will result in NaNs (:gh:`190` by `Fede Raimondo`_).
+
+- Expose `allow_overlap` parameter in  :class:`junifer.markers.SphereAgregation` and related markers (:gh:`190` by `Fede Raimondo`_).
```
Comment (synchon) [inline docs/changes/latest.inc:56]:
```
``allow_overlap``
```

File: junifer/external/nilearn/junifer_nifti_spheres_masker.py [hunk 2]
```diff
@@ -59,6 +64,155 @@
 """
 
 
+def _apply_mask_and_get_affinity(
```
Comment (synchon) [inline junifer/external/nilearn/junifer_nifti_spheres_masker.py:67]: This is for adding the custom check of empty spheres right?

File: junifer/stats.py [hunk 3]
```diff
@@ -82,6 +91,24 @@ def get_aggfunc_by_name(
...
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Data to count elements on.
+    axis : int, optional
+        The axis to count elements on (default 0).
+
+    Returns
+    -------
+    numpy.ndarray
```
Comment (synchon) [inline junifer/stats.py:106]: Should be an `int`.

**Review by synchon — CHANGES_REQUESTED** [review]

File: junifer/external/nilearn/junifer_nifti_spheres_masker.py (hunk 2 above)
Comment (fraimondo) [reply junifer/external/nilearn/junifer_nifti_spheres_masker.py:67]: Indeed the only change from nilearn was to remove the check.

**Comment by fraimondo** [comment]
done
can't test more without writing nilearn tests

File: junifer/external/nilearn/junifer_nifti_spheres_masker.py (hunk 2 above)
Comment (synchon) [reply junifer/external/nilearn/junifer_nifti_spheres_masker.py:67]: Sounds good.

**Review by synchon — APPROVED** [review]

**Pull request merged**